src/snapshot_middleware/yaml.rs

```diff
@@ -0,0 +1,116 @@
+use std::path::Path;
+
+use anyhow::Context;
+use maplit::hashmap;
+use memofs::{IoResultExt, Vfs};
+
+use crate::{
+    lua_ast::{Expression, Statement},
+    snapshot::{InstanceContext, InstanceMetadata, InstanceSnapshot},
+};
+
+use super::meta_file::AdjacentMetadata;
+
+pub fn snapshot_yaml(
+    context: &InstanceContext,
+    vfs: &Vfs,
+    path: &Path,
+    name: &str,
+) -> anyhow::Result<Option<InstanceSnapshot>> {
+    let contents = vfs.read(path)?;
+
+    let value: serde_yaml::Value = serde_yaml::from_slice(&contents)
+        .with_context(|| format!("File contains malformed YAML: {}", path.display()))?;
+
+    let as_lua = yaml_to_lua(value).to_string();
+
+    let properties = hashmap! {
```

Comment:
`hashmap!` doesn't really need to exist anymore in Rust 2021.

```rs
HashMap::from_iter([
    ("Source".to_owned(), as_lua.into()),
]),
```

...works, and has the benefit of not needing a macro